### Year 4 Topology 5 - Limit Checking - Overloading, Voltage limits

In [1]:
import pandas as pd
import altair as alt

##### Loading

##### Percentage Loading in Branches 

Calculation related can be found in PSSE PAGVI Manual Chapter 6 Power system network simulations - Section 6.6.7 Activity Rate 

%I = MVA(actual)/(MVA(rated)*Vpu)

Vpu -> Bus Voltage at from end 

In [2]:
list_all = []
list_gens = [0,50,100]
list_lsc = ['LLS','RLS','HLS']
list_gen_hydro = [400,500,600] #dry, average, wet hydological scenarios
list_gen_wind = [0,40,75] # for 3 wind gust scenarios
for gen in list_gens:
    for gen_hy in list_gen_hydro:
        for gen_wi in list_gen_wind:
            for lsc in list_lsc:
                file_csv_out = 'load\savnw_sol_' + str(gen) +'_hy_' + str(gen_hy) +  '_wi_' + str(gen_wi) + '_' + str(lsc) + '_l.csv'
                data_load = pd.read_csv(file_csv_out,header=1, skiprows=0)
                data_fil = data_load
                data_fil['From Bus V'] = data_fil['From Bus Ext'].str.extract(r'(\d+\.\d+)') 
                data_fil['To Bus V'] = data_fil['To Bus Ext'].str.extract(r'(\d+\.\d+)')
                data_fil['Scenario']= 'Solar = ' + str(gen) + ' MW, ' + 'Hydro = ' + str(gen_hy) + ' MW, ' + 'Wind = ' + str(gen_wi) + ' MW, '  + lsc.upper()
                list_all.append(data_fil)
data_all = pd.concat(list_all).reset_index(drop=True)
data_all = data_all[['Scenario','From Bus', 'From Bus Ext', 'To Bus', 'To Bus Ext', 'Ckt ID',
       'Non Metered Bus', ' Branch I', 'Branch Flow MW', 'Branch Flow MVAR',
       'Branch MVA', 'MW Loss', 'MVAR Loss', 'Rate', '%I', '%MVA',
       'From Bus V', 'To Bus V']]

##### Line and Transformer Loading

Transformer and branch loading are categorised to following brackets based on the range to which it falls into 
* Loading less than 25%
* Loading 25% to 50%
* Loading 50% to 70%
* Loading 70% to 80%
* Loading 80% to 85%
* Loading 85% to 90%
* Loading 90% to 95%
* Loading 95% to 100%
* Loading 100% and above

Separating transformer and line branch loadings

In [3]:
transformer_loading = data_all[data_all['From Bus V']!=data_all['To Bus V']].reset_index(drop=True)
branch_loading = data_all[data_all['From Bus V'] == data_all['To Bus V']].reset_index(drop=True)

Defining loading brackets for transformer and line branches 

In [4]:
data = transformer_loading[['Scenario','From Bus', 'From Bus Ext', 'To Bus', 'To Bus Ext', 'Ckt ID','Branch Flow MW', 'Branch Flow MVAR','Branch MVA', 'Rate','%MVA']]
data['Loading'] = ''
data.loc[(data['%MVA'] >100), 'Loading'] = '%MVA>100'
data.loc[(data['%MVA'] <100), 'Loading'] = '95<%MVA<100'
data.loc[(data['%MVA'] <95), 'Loading'] = '90<%MVA<95'
data.loc[(data['%MVA'] <90), 'Loading'] = '85<%MVA<90'
data.loc[(data['%MVA'] <85), 'Loading'] = '80<%MVA<85'
data.loc[(data['%MVA'] <80), 'Loading'] = '70<%MVA<80'
data.loc[(data['%MVA'] <70), 'Loading'] = '50<%MVA<70'
data.loc[(data['%MVA'] <50), 'Loading'] = '25<%MVA<50'
data.loc[(data['%MVA'] <25), 'Loading'] = '%MVA<25'

<ipython-input-4-7f16dd71236d>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['Loading'] = ''


In [5]:
data_l = branch_loading[['Scenario','From Bus', 'From Bus Ext', 'To Bus', 'To Bus Ext', 'Ckt ID','Branch Flow MW', 'Branch MVA', 'Rate','%I']]
data_l['Loading'] = ''
data_l.loc[(data_l['%I'] >100), 'Loading'] = '%I>100'
data_l.loc[(data_l['%I'] <100), 'Loading'] = '95<%I<100'
data_l.loc[(data_l['%I'] <95), 'Loading'] = '90<%I<95'
data_l.loc[(data_l['%I'] <90), 'Loading'] = '85<%I<90'
data_l.loc[(data_l['%I'] <85), 'Loading'] = '80<%I<85'
data_l.loc[(data_l['%I'] <80), 'Loading'] = '70<%I<80'
data_l.loc[(data_l['%I'] <70), 'Loading'] = '50<%I<70'
data_l.loc[(data_l['%I'] <50), 'Loading'] = '25<%I<50'
data_l.loc[(data_l['%I'] <25), 'Loading'] = '%I<25'

<ipython-input-5-abcf08d21c83>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_l['Loading'] = ''


##### Histogram Transformer Branch Loading 

A histogram for each scenario is plotted to check the no of transformer branches in the above defined loading bracket 

In [6]:
alt.Chart(data).mark_bar().encode(
    alt.Y('Loading',scale = alt.Scale(domain=['%MVA>100','95<%MVA<100','90<%MVA<95','85<%MVA<90','80<%MVA<85','70<%MVA<80','50<%MVA<70','25<%MVA<50','%MVA<25'])),
    alt.X('count()')).properties(height = 120, width = 250).facet('Scenario',columns=5).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

It can be seen that for most of scenarios the transformer branch loading falls into a bracket of loading between 25 to 50 percentage. 

##### Histogram Line Branch Loading 

A histogram is plotted to check the no of line branches in the above defined loading bracket 

In [7]:
alt.Chart(data_l).mark_bar().encode(
    alt.Y('Loading', scale =alt.Scale(domain=['%I>100','95<%I<100','90<%I<95','85<%I<90','80<%I<85','70<%I<80','50<%I<70','25<%I<50','%I<25'])),
    alt.X('count()')).properties(height = 120, width = 250).facet('Scenario',columns=5).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

It can be seen that for most of scenarios the line branch loading falls into a bracket of loading between 25% to 50%

In the following section the lines and transformer branches with loading greater than 80% is filtered 

##### Checking transformer loading greater than 80% 

In [8]:
transformer_overload  = transformer_loading[transformer_loading['%MVA'] > 80.0].reset_index(drop=True)
transformer_overload['Branch'] = transformer_overload['From Bus'].astype(str) + ' ' + transformer_overload['From Bus Ext'] + ' to ' + \
transformer_overload['To Bus'].astype(str) + ' '+ transformer_overload['To Bus Ext'] + ' ' + transformer_overload['Ckt ID'].astype(str)
list_branch = transformer_overload['Branch'].unique()
print(list_branch)

['3018 CATDOG_G    13.800 to 3008 CATDOG      230.00 1'
 '206 URBGEN      18.000 to 205 SUB230      230.00 1']


In [9]:
transformer_overload

Scenario  From Bus  \
0      Solar = 0 MW, Hydro = 400 MW, Wind = 0 MW, LLS      3018   
1      Solar = 0 MW, Hydro = 400 MW, Wind = 0 MW, RLS      3018   
2      Solar = 0 MW, Hydro = 400 MW, Wind = 0 MW, HLS       206   
3      Solar = 0 MW, Hydro = 400 MW, Wind = 0 MW, HLS      3018   
4     Solar = 0 MW, Hydro = 400 MW, Wind = 40 MW, LLS      3018   
..                                                ...       ...   
85  Solar = 100 MW, Hydro = 600 MW, Wind = 40 MW, RLS      3018   
86  Solar = 100 MW, Hydro = 600 MW, Wind = 40 MW, HLS      3018   
87  Solar = 100 MW, Hydro = 600 MW, Wind = 75 MW, LLS      3018   
88  Solar = 100 MW, Hydro = 600 MW, Wind = 75 MW, RLS      3018   
89  Solar = 100 MW, Hydro = 600 MW, Wind = 75 MW, HLS      3018   

          From Bus Ext  To Bus          To Bus Ext  Ckt ID  Non Metered Bus  \
0   CATDOG_G    13.800    3008  CATDOG      230.00       1             3018   
1   CATDOG_G    13.800    3008  CATDOG      230.00       1             3018   
2   URBGEN      18.000     205  SUB230      230.00       1              206   
3   CATDOG_G    13.800    3008  CATDOG      230.00       1             3018   
4   CATDOG_G    13.800    3008  CATDOG      230.00       1             3018   
..                 ...     ...                 ...     ...              ...   
85  CATDOG_G    13.800    3008  CATDOG      230.00       1             3018   
86  CATDOG_G    13.800    3008  CATDOG      230.00       1             3018   
87  CATDOG_G    13.800    3008  CATDOG      230.00       1             3018   
88  CATDOG_G    13.800    3008  CATDOG      230.00       1             3018   
89  CATDOG_G    13.800    3008  CATDOG      230.00       1             3018   

     Branch I  Branch Flow MW  Branch Flow MVAR  Branch MVA  MW Loss  \
0        4792           90.00              80.0      120.42     0.03   
1        4807           90.00              80.0      120.42     0.03   
2       29639          744.11             600.0      955.88     2.22   
3        4886           90.00              80.0      120.41     0.03   
4        4791           90.00              80.0      120.42     0.03   
..        ...             ...               ...         ...      ...   
85       4848           90.00              80.0      120.41     0.03   
86       4934           90.00              80.0      120.41     0.03   
87       4797           90.00              80.0      120.42     0.03   
88       4850           90.00              80.0      120.41     0.03   
89       4936           90.00              80.0      120.41     0.03   

    MVAR Loss    Rate     %I   %MVA From Bus V To Bus V  \
0       11.15   150.0  76.36  80.28     13.800   230.00   
1       11.23   150.0  76.61  80.28     13.800   230.00   
2      113.82  1150.0  80.35  83.12     18.000   230.00   
3       11.59   150.0  77.86  80.28     13.800   230.00   
4       11.15   150.0  76.35  80.28     13.800   230.00   
..        ...     ...    ...    ...        ...      ...   
85      11.42   150.0  77.27  80.28     13.800   230.00   
86      11.83   150.0  78.64  80.28     13.800   230.00   
87      11.18   150.0  76.44  80.28     13.800   230.00   
88      11.43   150.0  77.29  80.28     13.800   230.00   
89      11.84   150.0  78.67  80.28     13.800   230.00   

                                               Branch  
0   3018 CATDOG_G    13.800 to 3008 CATDOG      23...  
1   3018 CATDOG_G    13.800 to 3008 CATDOG      23...  
2   206 URBGEN      18.000 to 205 SUB230      230....  
3   3018 CATDOG_G    13.800 to 3008 CATDOG      23...  
4   3018 CATDOG_G    13.800 to 3008 CATDOG      23...  
..                                                ...  
85  3018 CATDOG_G    13.800 to 3008 CATDOG      23...  
86  3018 CATDOG_G    13.800 to 3008 CATDOG      23...  
87  3018 CATDOG_G    13.800 to 3008 CATDOG      23...  
88  3018 CATDOG_G    13.800 to 3008 CATDOG      23...  
89  3018 CATDOG_G    13.800 to 3008 CATDOG      23...  

[90 rows x 19 columns]

##### Transformer branch overload plot

In [10]:
alt.Chart(transformer_overload).mark_line().encode(
    alt.X('%MVA', scale=alt.Scale(zero=False)),
    alt.Y('Scenario',axis = alt.Axis(labelLimit= 250, titleX=-260), sort='-x'),
    alt.Color('Branch', legend=None)).properties(height = 700, width =400).facet('Branch',columns=2).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

##### Checking branch loading greater than 80% 

In [11]:

branch_overload = branch_loading[branch_loading['%I'] > 80.0].reset_index(drop=True)
branch_overload['Branch'] = branch_overload['From Bus'].astype(str) + ' ' + branch_overload['From Bus Ext'] + ' to ' + \
branch_overload['To Bus'].astype(str) + ' '+ branch_overload['To Bus Ext'] + ' ' + branch_overload['Ckt ID'].astype(str)
list_branch = branch_overload['Branch'].unique()
print(list_branch)


['154 DOWNTN      230.00 to 203 EAST230     230.00 1'
 '203 EAST230     230.00 to 154 DOWNTN      230.00 1'
 '153 MID230      230.00 to 154 DOWNTN      230.00 1'
 '154 DOWNTN      230.00 to 153 MID230      230.00 1'
 '203 EAST230     230.00 to 205 SUB230      230.00 1'
 '203 EAST230     230.00 to 205 SUB230      230.00 2'
 '205 SUB230      230.00 to 203 EAST230     230.00 1'
 '205 SUB230      230.00 to 203 EAST230     230.00 2']


In [12]:
branch_overload

Scenario  From Bus  \
0       Solar = 0 MW, Hydro = 400 MW, Wind = 0 MW, LLS       154   
1       Solar = 0 MW, Hydro = 400 MW, Wind = 0 MW, RLS       154   
2       Solar = 0 MW, Hydro = 400 MW, Wind = 0 MW, RLS       203   
3       Solar = 0 MW, Hydro = 400 MW, Wind = 0 MW, HLS       154   
4       Solar = 0 MW, Hydro = 400 MW, Wind = 0 MW, HLS       203   
..                                                 ...       ...   
146  Solar = 100 MW, Hydro = 600 MW, Wind = 75 MW, LLS       203   
147  Solar = 100 MW, Hydro = 600 MW, Wind = 75 MW, RLS       154   
148  Solar = 100 MW, Hydro = 600 MW, Wind = 75 MW, RLS       203   
149  Solar = 100 MW, Hydro = 600 MW, Wind = 75 MW, HLS       154   
150  Solar = 100 MW, Hydro = 600 MW, Wind = 75 MW, HLS       203   

           From Bus Ext  To Bus          To Bus Ext  Ckt ID  Non Metered Bus  \
0    DOWNTN      230.00     203  EAST230     230.00       1              203   
1    DOWNTN      230.00     203  EAST230     230.00       1              203   
2    EAST230     230.00     154  DOWNTN      230.00       1              203   
3    DOWNTN      230.00     203  EAST230     230.00       1              203   
4    EAST230     230.00     154  DOWNTN      230.00       1              203   
..                  ...     ...                 ...     ...              ...   
146  EAST230     230.00     154  DOWNTN      230.00       1              203   
147  DOWNTN      230.00     203  EAST230     230.00       1              203   
148  EAST230     230.00     154  DOWNTN      230.00       1              203   
149  DOWNTN      230.00     203  EAST230     230.00       1              203   
150  EAST230     230.00     154  DOWNTN      230.00       1              203   

      Branch I  Branch Flow MW  Branch Flow MVAR  Branch MVA  MW Loss  \
0          402         -147.76            -50.04      156.01     1.01   
1          411         -151.59            -48.78      159.25     1.06   
2          404          152.65             49.68      160.53     1.06   
3          434         -154.29            -57.02      164.49     1.18   
4          426          155.46             59.46      166.45     1.18   
..         ...             ...               ...         ...      ...   
146        442          169.95             45.49      175.93     1.26   
147        466         -172.17            -46.54      178.35     1.37   
148        460          173.54             50.73      180.80     1.37   
149        491         -174.81            -55.02      183.26     1.51   
150        483          176.32             61.04      186.59     1.51   

     MVAR Loss   Rate     %I   %MVA From Bus V To Bus V  \
0        10.09  200.0  80.18  78.00     230.00   230.00   
1        10.58  200.0  82.02  79.62     230.00   230.00   
2        10.58  200.0  80.51  80.27     230.00   230.00   
3        11.77  200.0  86.56  82.24     230.00   230.00   
4        11.77  200.0  84.86  83.22     230.00   230.00   
..         ...    ...    ...    ...        ...      ...   
146      12.65  200.0  88.23  87.96     230.00   230.00   
147      13.65  200.0  92.98  89.17     230.00   230.00   
148      13.65  200.0  91.66  90.40     230.00   230.00   
149      15.10  200.0  97.82  91.63     230.00   230.00   
150      15.10  200.0  96.32  93.29     230.00   230.00   

                                                Branch  
0    154 DOWNTN      230.00 to 203 EAST230     230....  
1    154 DOWNTN      230.00 to 203 EAST230     230....  
2    203 EAST230     230.00 to 154 DOWNTN      230....  
3    154 DOWNTN      230.00 to 203 EAST230     230....  
4    203 EAST230     230.00 to 154 DOWNTN      230....  
..                                                 ...  
146  203 EAST230     230.00 to 154 DOWNTN      230....  
147  154 DOWNTN      230.00 to 203 EAST230     230....  
148  203 EAST230     230.00 to 154 DOWNTN      230....  
149  154 DOWNTN      230.00 to 203 EAST230     230....  
150  203 EAST230     230.00 to 154 DOWNTN      230.... 

##### Transmission line branch overload plots

In [13]:
branch_fil = branch_overload[(branch_overload['Branch'] !='154 DOWNTN      230.00 to 203 EAST230     230.00 1') & (branch_overload['Branch'] !='203 EAST230     230.00 to 154 DOWNTN      230.00 1') ].reset_index(drop = True)
branch_fil_1 = branch_overload[(branch_overload['Branch'] =='154 DOWNTN      230.00 to 203 EAST230     230.00 1') | (branch_overload['Branch'] =='203 EAST230     230.00 to 154 DOWNTN      230.00 1') ].reset_index(drop = True)

In [14]:
alt.Chart(branch_fil).mark_line().encode(
    alt.X('%I', scale=alt.Scale(zero=False)),
    alt.Y('Scenario',axis = alt.Axis(labelLimit= 250, titleX=-260), sort='-x'),
    alt.Color('Branch', legend=None)).properties(height = 55, width =200).facet('Branch',columns=2).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

In [15]:
alt.Chart(branch_fil_1).mark_line().encode(
    alt.X('%I', scale=alt.Scale(zero=False)),
    alt.Y('Scenario',axis = alt.Axis(labelLimit= 250, titleX=-260), sort='-x'),
    alt.Color('Branch', legend=None)).properties(height = 550, width =200).facet('Branch',columns=3).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

##### Bus Voltage

In [16]:
list_volt = []
list_gens = [0,50,100]
list_lsc = ['LLS','RLS','HLS']
list_gen_hydro = [400,500,600] #dry, average, wet hydological scenarios
list_gen_wind = [0,40,75] # for 3 wind gust scenarios
for gen in list_gens:
    for gen_hy in list_gen_hydro:
        for gen_wi in list_gen_wind:
            for lsc in list_lsc:
                file_csv_out = 'voltage/savnw_sol_' + str(gen) +'_hy_' + str(gen_hy) +  '_wi_' + str(gen_wi) + '_' + str(lsc) + '_volt.csv'
                data_volt = pd.read_csv(file_csv_out,header=1, skiprows=0)
                data_volt['Scenario']= 'Solar = ' + str(gen) + ' MW, ' + 'Hydro = ' + str(gen_hy) + ' MW, ' + 'Wind = ' + str(gen_wi) + ' MW, '  + lsc.upper()
                list_volt.append(data_volt)
data_all_volt = pd.concat(list_volt).reset_index(drop=True)
data_all_volt = data_all_volt[['Scenario','Bus Number', 'Bus Name', 'Area', 'Base Voltage', 'Bus Voltage(PU)','Bus Voltage(kV)']]

Similar to branch loading categorization into brackets of loadings, voltages are also categorised into brackets

In [17]:
data_v = data_all_volt
data_v['Voltage Range'] = ''
data_v.loc[(data_v['Bus Voltage(PU)'] >1.1), 'Voltage Range'] = 'Vbus>1.1'
data_v.loc[(data_v['Bus Voltage(PU)'] <1.1), 'Voltage Range'] = '1.05<Vbus<1.1'
data_v.loc[(data_v['Bus Voltage(PU)'] <1.05), 'Voltage Range'] = '1.0<Vbus<1.05'
data_v.loc[(data_v['Bus Voltage(PU)'] <1.0), 'Voltage Range'] = '0.95<Vbus<1.0'
data_v.loc[(data_v['Bus Voltage(PU)'] <0.95), 'Voltage Range'] = '0.95<Vbus<0.9'
data_v.loc[(data_v['Bus Voltage(PU)'] <0.9), 'Voltage Range'] = 'Vbus<0.9'

##### Histogram Bus Voltages

A histogram is plotted to check the no of Buses in each of the above defined brackets 

In [18]:
alt.Chart(data_v).mark_bar().encode(
    alt.Y('Voltage Range', scale =alt.Scale(domain=['Vbus>1.1','1.05<Vbus<1.1','1.0<Vbus<1.05','0.95<Vbus<1.0','0.95<Vbus<0.9','Vbus<0.9'])),
    alt.X('count()')).properties(height = 100,width = 200).facet('Scenario',columns=6).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

It can be seen that most of the buses have voltages in the bracket of 1.0 PU and 1.05 PU

In the following section, the buses out of upper and lower normal operating range is filtered and reported 

##### Voltage Upper Limits 

In [19]:
data_all_volt[data_all_volt['Bus Voltage(PU)']>1.05].reset_index(drop=True)['Bus Number'].unique()

array([3018,  206,  211,  212], dtype=int64)

In [20]:
data_high = data_all_volt[data_all_volt['Bus Voltage(PU)']>1.05].reset_index(drop=True)
data_high

Scenario  Bus Number  \
0      Solar = 0 MW, Hydro = 400 MW, Wind = 0 MW, LLS        3018   
1      Solar = 0 MW, Hydro = 400 MW, Wind = 0 MW, RLS         206   
2     Solar = 0 MW, Hydro = 400 MW, Wind = 40 MW, LLS        3018   
3     Solar = 0 MW, Hydro = 400 MW, Wind = 40 MW, RLS         206   
4     Solar = 0 MW, Hydro = 400 MW, Wind = 75 MW, LLS        3018   
..                                                ...         ...   
65   Solar = 100 MW, Hydro = 600 MW, Wind = 0 MW, HLS         212   
66  Solar = 100 MW, Hydro = 600 MW, Wind = 40 MW, HLS         211   
67  Solar = 100 MW, Hydro = 600 MW, Wind = 40 MW, HLS         212   
68  Solar = 100 MW, Hydro = 600 MW, Wind = 75 MW, HLS         211   
69  Solar = 100 MW, Hydro = 600 MW, Wind = 75 MW, HLS         212   

        Bus Name  Area  Base Voltage  Bus Voltage(PU)  Bus Voltage(kV)  \
0   CATDOG_G         5          13.8            1.051           14.508   
1   URBGEN           2          18.0            1.053           18.948   
2   CATDOG_G         5          13.8            1.051           14.510   
3   URBGEN           2          18.0            1.053           18.945   
4   CATDOG_G         5          13.8            1.051           14.508   
..           ...   ...           ...              ...              ...   
65  HYDRO_N          2          20.0            1.063           21.257   
66  HYDRO_G          2          20.0            1.063           21.253   
67  HYDRO_N          2          20.0            1.063           21.253   
68  HYDRO_G          2          20.0            1.063           21.260   
69  HYDRO_N          2          20.0            1.063           21.260   

    Voltage Range  
0   1.05<Vbus<1.1  
1   1.05<Vbus<1.1  
2   1.05<Vbus<1.1  
3   1.05<Vbus<1.1  
4   1.05<Vbus<1.1  
..            ...  
65  1.05<Vbus<1.1  
66  1.05<Vbus<1.1  
67  1.05<Vbus<1.1  
68  1.05<Vbus<1.1  
69  1.05<Vbus<1.1  

[70 rows x 8 columns]

##### Upper voltage limit violation plot


In [21]:
alt.Chart(data_high).mark_line().encode(
    alt.X('Bus Voltage(PU)', scale=alt.Scale(zero=False)),
    alt.Y('Scenario',axis = alt.Axis(labelLimit= 250, titleX=-260), sort='-x'),
    alt.Color('Bus Number:N', legend=None)).properties(height = 200, width = 280).facet('Bus Number:N',columns=2).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

##### Voltage Lower Limits

In [22]:
data_all_volt[data_all_volt['Bus Voltage(PU)']<0.95]['Bus Number'].unique()

array([154, 205], dtype=int64)

In [23]:
data_low = data_all_volt[data_all_volt['Bus Voltage(PU)']<0.95].reset_index(drop=True)
data_low

Scenario  Bus Number  \
0      Solar = 0 MW, Hydro = 500 MW, Wind = 0 MW, HLS         154   
1     Solar = 0 MW, Hydro = 500 MW, Wind = 40 MW, HLS         154   
2     Solar = 0 MW, Hydro = 500 MW, Wind = 75 MW, HLS         154   
3      Solar = 0 MW, Hydro = 600 MW, Wind = 0 MW, HLS         154   
4      Solar = 0 MW, Hydro = 600 MW, Wind = 0 MW, HLS         205   
5     Solar = 0 MW, Hydro = 600 MW, Wind = 40 MW, HLS         154   
6     Solar = 0 MW, Hydro = 600 MW, Wind = 40 MW, HLS         205   
7     Solar = 0 MW, Hydro = 600 MW, Wind = 75 MW, HLS         154   
8     Solar = 0 MW, Hydro = 600 MW, Wind = 75 MW, HLS         205   
9     Solar = 50 MW, Hydro = 500 MW, Wind = 0 MW, HLS         154   
10   Solar = 50 MW, Hydro = 500 MW, Wind = 40 MW, HLS         154   
11   Solar = 50 MW, Hydro = 500 MW, Wind = 75 MW, HLS         154   
12    Solar = 50 MW, Hydro = 600 MW, Wind = 0 MW, HLS         154   
13    Solar = 50 MW, Hydro = 600 MW, Wind = 0 MW, HLS         205   
14   Solar = 50 MW, Hydro = 600 MW, Wind = 40 MW, HLS         154   
15   Solar = 50 MW, Hydro = 600 MW, Wind = 40 MW, HLS         205   
16   Solar = 50 MW, Hydro = 600 MW, Wind = 75 MW, HLS         154   
17   Solar = 50 MW, Hydro = 600 MW, Wind = 75 MW, HLS         205   
18   Solar = 100 MW, Hydro = 500 MW, Wind = 0 MW, HLS         154   
19  Solar = 100 MW, Hydro = 500 MW, Wind = 40 MW, HLS         154   
20  Solar = 100 MW, Hydro = 500 MW, Wind = 75 MW, HLS         154   
21   Solar = 100 MW, Hydro = 600 MW, Wind = 0 MW, HLS         154   
22   Solar = 100 MW, Hydro = 600 MW, Wind = 0 MW, HLS         205   
23  Solar = 100 MW, Hydro = 600 MW, Wind = 40 MW, HLS         154   
24  Solar = 100 MW, Hydro = 600 MW, Wind = 40 MW, HLS         205   
25  Solar = 100 MW, Hydro = 600 MW, Wind = 75 MW, HLS         154   
26  Solar = 100 MW, Hydro = 600 MW, Wind = 75 MW, HLS         205   

        Bus Name  Area  Base Voltage  Bus Voltage(PU)  Bus Voltage(kV)  \
0   DOWNTN           1         230.0            0.942          216.551   
1   DOWNTN           1         230.0            0.942          216.608   
2   DOWNTN           1         230.0            0.941          216.499   
3   DOWNTN           1         230.0            0.929          213.689   
4   SUB230           2         230.0            0.939          215.956   
5   DOWNTN           1         230.0            0.929          213.744   
6   SUB230           2         230.0            0.939          216.006   
7   DOWNTN           1         230.0            0.929          213.622   
8   SUB230           2         230.0            0.939          215.891   
9   DOWNTN           1         230.0            0.946          217.494   
10  DOWNTN           1         230.0            0.946          217.551   
11  DOWNTN           1         230.0            0.945          217.446   
12  DOWNTN           1         230.0            0.934          214.746   
13  SUB230           2         230.0            0.943          216.986   
14  DOWNTN           1         230.0            0.934          214.802   
15  SUB230           2         230.0            0.944          217.037   
16  DOWNTN           1         230.0            0.933          214.684   
17  SUB230           2         230.0            0.943          216.926   
18  DOWNTN           1         230.0            0.949          218.164   
19  DOWNTN           1         230.0            0.949          218.223   
20  DOWNTN           1         230.0            0.948          218.120   
21  DOWNTN           1         230.0            0.937          215.510   
22  SUB230           2         230.0            0.947          217.743   
23  DOWNTN           1         230.0            0.937          215.566   
24  SUB230           2         230.0            0.947          217.795   
25  DOWNTN           1         230.0            0.937          215.453   
26  SUB230           2         230.0            0.946          217.688   

    Voltage Range  
0   0.95<Vbus<0.9  

##### Lower voltage limit violation plot

In [24]:
alt.Chart(data_low).mark_line().encode(
    alt.X('Bus Voltage(PU)', scale=alt.Scale(zero=False)),
    alt.Y('Scenario',axis = alt.Axis(labelLimit= 250, titleX=-260), sort='x'),
    alt.Color('Bus Number:N', legend=None)).properties(height = 200, width = 280).facet('Bus Number:N',columns=2).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

#### Conclusion 

Year 4, Toplogy 5 limit violation check for overloading and voltage was carried out for the three load scenarios. It was seen that
1. Transformer and branch overloading of greater than 80% was reported for the following branches
   * Transformer 3018 CATDOG_G    13.800 to 3008 CATDOG      230.00 1
   * Transformer 206 URBGEN      18.000 to 205 SUB230      230.00 1
   * Branch 154 DOWNTN      230.00 to 203 EAST230     230.00 1
   * Branch 203 EAST230     230.00 to 154 DOWNTN      230.00 1
   * Branch 153 MID230      230.00 to 154 DOWNTN      230.00 1
   * Branch 54 DOWNTN      230.00 to 153 MID230      230.00 1
   * Branch 203 EAST230     230.00 to 205 SUB230      230.00 1
   * Branch 203 EAST230     230.00 to 205 SUB230      230.00 2
   * Branch 205 SUB230      230.00 to 203 EAST230     230.00 1
   * Branch 205 SUB230      230.00 to 203 EAST230     230.00 2
2. The buses violating the upper voltage limit are
   * 3018, 206, 211, 212
3. The buses violating the lower voltage limit are
   * 154, 205   